hazelcast/src/main/java/com/hazelcast/spi/impl/servicemanager/impl/ServiceManagerImpl.java

```diff
@@ -274,7 +274,7 @@ private void shutdownService(final ManagedService service, final boolean termina
         }
     }
 
-    private synchronized void registerService(String serviceName, Object service) {
+    protected synchronized void registerService(String serviceName, Object service) {
```

Comment:
Hazelcast currently has a way to register user defined services via XML (http://docs.hazelcast.org/docs/3.5/manual/html-single/index.html#user-defined-services). If Hazelcast provides, similar way with programmatic API, you don't need to make this change by opening the class for extension.